sources/TileDB.CSharp/Marshalling/MarshaledContiguousStringCollection.cs

```diff
@@ -21,7 +21,7 @@ public MarshaledContiguousStringCollection(IReadOnlyCollection<string> strings,
         Encoding encoding = MarshaledString.GetEncoding(dataType);
         try
         {
-            Offsets = (ulong*)Marshal.AllocHGlobal(strings.Count * sizeof(ulong));
+            Offsets = (ulong*)NativeMemory.Alloc((nuint)strings.Count, sizeof(ulong));
```

Comment:
For historical reasons, [`Marshal.AllocHGlobal`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.allochglobal) uses a legacy and slower API on Windows to allocate memory. [`NativeMemory.Alloc`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.nativememory.alloc?view=net-9.0) was introduced more recently and is documented to always call `malloc` while also featuring slightly improved ergonomics.